cpp/src/IceGrid/SessionServantManager.h

```diff
@@ -22,51 +22,44 @@ namespace IceGrid
             const std::string&,
             bool,
             const std::string&,
-            const std::shared_ptr<Ice::Object>&,
+            const Ice::ObjectPtr&,
             const std::string&,
-            const std::shared_ptr<Ice::Object>&,
+            const Ice::ObjectPtr&,
             const std::string&,
-            const std::shared_ptr<Ice::Object>&,
+            const Ice::ObjectPtr&,
             const std::shared_ptr<AdminCallbackRouter>&);
 
-        std::shared_ptr<Ice::Object> locate(const Ice::Current&, std::shared_ptr<void>&);
-        void finished(const Ice::Current&, const std::shared_ptr<Ice::Object>&, const std::shared_ptr<void>&);
+        Ice::ObjectPtr locate(const Ice::Current&, std::shared_ptr<void>&);
+        void finished(const Ice::Current&, const Ice::ObjectPtr&, const std::shared_ptr<void>&);
         void deactivate(const std::string&);
 
-        Ice::ObjectPrx
-        addSession(const std::shared_ptr<Ice::Object>&, const std::shared_ptr<Ice::Connection>&, const std::string&);
-        void setSessionControl(
-            const std::shared_ptr<Ice::Object>&,
-            const Glacier2::SessionControlPrx&,
-            const Ice::IdentitySeq&);
-        std::optional<Glacier2::IdentitySetPrx> getGlacier2IdentitySet(const std::shared_ptr<Ice::Object>&);
-        std::optional<Glacier2::StringSetPrx> getGlacier2AdapterIdSet(const std::shared_ptr<Ice::Object>&);
-        void removeSession(const std::shared_ptr<Ice::Object>&);
-
-        Ice::ObjectPrx add(const std::shared_ptr<Ice::Object>&, const std::shared_ptr<Ice::Object>&);
+        Ice::ObjectPrx addSession(const Ice::ObjectPtr&, const std::shared_ptr<Ice::Connection>&, const std::string&);
```

Comment:
We should do the same with ` std::shared_ptr<Ice::Connection>` for another PR!

Reply:
Yes, and many other classes.